compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1307,15 +1310,14 @@ pub fn catch_with_exit_code(f: impl FnOnce() -> interface::Result<()>) -> i32 {
 static ICE_PATH: OnceLock<Option<PathBuf>> = OnceLock::new();
 
 fn ice_path() -> &'static Option<PathBuf> {
+    ice_path_with_config(None)
+}
+
+fn ice_path_with_config(config: Option<&UnstableOptions>) -> &'static Option<PathBuf> {
     ICE_PATH.get_or_init(|| {
         if !rustc_feature::UnstableFeatures::from_environment(None).is_nightly_build() {
             return None;
         }
-        if let Some(s) = std::env::var_os("RUST_BACKTRACE")
-            && s == "0"
-        {
-            return None;
-        }
```

Comment:
this is unrelated to the rest of the feature. I've removed it because it was causing unintentional behavior (disabling backtraces were completely disabling saving ICEs, instead of just removing the backtrace output from the ICE report)